src/transaction/log_checkpoint_info.cpp

```diff
@@ -286,5 +287,164 @@ namespace cublog
     log_system_tdes::map_all_tdes (mapper);
   }
 
+  void
+  checkpoint_info::recovery_analysis (THREAD_ENTRY *thread_p, log_lsa &start_redo_lsa)
+  {
+    int i, size, error_code;
+    void *area;
+    LOG_TDES *tdes;
+    LOG_INFO_CHKPT_TRANS *chkpt_trans;
+    LOG_INFO_CHKPT_TRANS *chkpt_one;
+    LOG_INFO_CHKPT_SYSOP *chkpt_topops;
+    LOG_INFO_CHKPT_SYSOP *chkpt_topone;
+    LOG_PAGE *log_page_local = NULL;
+    LOG_LSA log_lsa_local;
+    char log_page_buffer[IO_MAX_PAGE_SIZE + MAX_ALIGNMENT];
+    LOG_REC_SYSOP_START_POSTPONE sysop_start_postpone;
```

Comment:
- Remove unused variables + unnecessary pointers chkpt_one/chkpt_topone (see other comments).
- Give variables an initial value to make sure they are not used uninitialized.
```suggestion
    int error_code;
    LOG_TDES *tdes = nullptr;
    LOG_PAGE *log_page_local = NULL;
    LOG_LSA log_lsa_local;
    char log_page_buffer[IO_MAX_PAGE_SIZE + MAX_ALIGNMENT];
    LOG_REC_SYSOP_START_POSTPONE sysop_start_postpone;
```